test/boost/sstable_compaction_test.cc

```diff
@@ -2339,6 +2340,88 @@ SEASTAR_THREAD_TEST_CASE(sstable_scrub_validate_mode_test_corrupted_file) {
     });
 }
 
+SEASTAR_THREAD_TEST_CASE(sstable_scrub_validate_mode_test_corrupted_file_digest) {
+    scrub_test_framework test(compress_sstable::yes);
+
+    auto schema = test.schema();
+
+    auto muts = tests::generate_random_mutations(
+            test.random_schema(),
+            tests::uncompactible_timestamp_generator(test.seed()),
+            tests::no_expiry_expiry_generator(),
+            std::uniform_int_distribution<size_t>(10, 10)).get();
+
+    test.run(schema, muts, [] (table_for_tests& table, compaction::table_state& ts, std::vector<sstables::shared_sstable> sstables) {
+        BOOST_REQUIRE(sstables.size() == 1);
+        auto sst = sstables.front();
+
+        // This test is about corrupted data with valid per-chunk checksums.
+        // This kind of corruption should be detected by the digest check.
+        // Triggering this is not trivial, so we corrupt the Digest file instead.
+        auto f = open_file_dma(sstables::test(sst).filename(component_type::Digest).native(), open_flags::rw).get();
+        auto stream = make_file_input_stream(f);
+        auto digest_str = util::read_entire_stream_contiguous(stream).get();
+        auto digest = boost::lexical_cast<uint32_t>(digest_str);
+        auto new_digest = to_sstring<bytes>(digest + 1); // a random invalid digest
+        f.dma_write(0, new_digest.c_str(), new_digest.size()).get();
+        stream.close().get();
```

Comment:
This is not exception safe. An exception thrown above will possibly result in a crash, because the stream is destroyed before close.
A better idiom is:
```cpp
auto stream = ...
auto close_stream = deferred_close(stream); // RAII closer, works in seastar threads contexts (SEASTAR_THREAD_TEST_CASE creates a seastar thread context)
```

Same for the tests below.